server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -93,7 +92,16 @@ public ContextIndexSearcher(
         QueryCachingPolicy queryCachingPolicy,
         boolean wrapWithExitableDirectoryReader
     ) throws IOException {
-        this(reader, similarity, queryCache, queryCachingPolicy, new MutableQueryTimeout(), wrapWithExitableDirectoryReader, null);
+        this(
+            reader,
+            similarity,
+            queryCache,
+            queryCachingPolicy,
+            new MutableQueryTimeout(),
+            Integer.MAX_VALUE,
+            wrapWithExitableDirectoryReader,
+            null
```

Comment:
Providing null is ok here?

Reply:
yes, that means no concurrency

Reply:
Also same confusion here.

Reply:
not related to this PR, but heads up: using null executor as an indication is fine for now, but as we are discussing always offloading to the executor, even for sequential execution, I think that we will have to consider always having an executor, and a separate way to optionally disable concurrency.